### ISSUE TITLE
Immediate-mode clientside 2D/3D rendering functions

### ISSUE THREAD

**Issue by philpax** [AmbientRun/Ambient#431]
Right now, the only way to render basic geometric shapes is to use the `Element`s, like I do with my racing game:

```rust
#[element_component]
fn DebugLines(hooks: &mut Hooks) -> Element {
    let lines = hooks.use_query(components::debug_lines());

    Group::el(lines.into_iter().flat_map(|(_, lines)| {
        lines
            .chunks(2)
            .map(|line| {
                let [start, end]: [Vec3; 2] = line.try_into().unwrap();

                Element::new()
                    .init_default(rect())
                    .with_default(main_scene())
                    .with(line_from(), start)
                    .with(line_to(), end)
                    .with(line_width(), 0.05)
                    .with(color(), vec4(1., 1., 1., 1.))
            })
            .collect::<Vec<_>>()
    }))
}
```

However, this requires having an `Element` workflow set up, which can be quite difficult to wire up to the game state / drive correctly. This is especially true if your rendering logic is  We should improve that experience where possible, but we should also consider adding primitives for basic imperative rendering:

- Rendering lines / rects
- Rendering text
- Rendering points

Where possible, we should encourage people to use `Element` or the dedicated renderer, so we should avoid offering *too* much functionality here. This is purely for convenience, so that you can easily debug in-place.

**Comment by VinhTruongAmbient** [comment]
Yes, this kind of facility very useful for debugging. Kinda like https://docs.unity3d.com/ScriptReference/Debug.DrawLine.html where you can call it anywhere in your client code and have a line.

**Comment by VinhTruongAmbient** [comment]
The simplest APIs would be to take one primitive at the time, but do you think the WIT calling overhead makes it unusable? If say I want to draw 1000 lines every frame. We can also make all APIs always take a slice so users can them more efficiently. Or both.

**Comment by philpax** [comment]
Good question. I'd say we should try a one function=one primitive solution to begin with, and then consider adding solutions for bulk primitives if the overhead becomes untenable.

**Comment by ten3roberts** [comment]
We already have the gizmos system which have an immediate mode line/sphere/rect/cube rendering.

Could we make use of that and extend it and expose it to the guest api?

**Comment by VinhTruongAmbient** [comment]
> We already have the gizmos system which have an immediate mode line/sphere/rect/cube rendering.
> 
> Could we make use of that and extend it and expose it to the guest api?

I looked at the Gizmos but felt like the design and implementation are a bit overkill for this very simple use case. Rather than trying to retrofit existing gizmos, I'd like to do a clean implementation.

**Comment by VinhTruongAmbient** [comment]
As an aside: gizmos usually mean something that you use to interact with an object inside an editor. I think that domain is where the gizmos belong.
